### PR TITLE
Add additional image size support to Post/Page Grid

### PR DESCRIPTION
**Pull request by mindctrl**
Adds registered image sizes to the Image Size dropdown.

Shows an indicator when the selected image size is not available for a post being shown in the grid.

To test, be sure to run `npm run build-assets` after checking out this branch.

Tasks remaining
- [x] Update [Featured Image Sizes](https://github.com/studiopress/atomic-blocks/wiki/Post-Grid-Block#featured-image-sizes) section of docs.
- [x] Warning icon doesn't show after save/reload, but does show on hover. Must be fixed to show warning icon on page load.
- [x] What do to as a fallback when the selected image size is no longer available?
- [x] Test Authority theme to ensure the selected image size is the one that is displayed.

### CODE REVIEW

File: src/blocks/block-post-grid/components/image.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+								<Dashicon
+									icon={ 'warning' }
+								/>
+							</div>
+
+							<Placeholder
+								className={ 'ab-post-grid-no-image-placeholder' }
+							>
+								<Dashicon icon={ 'info' } />
+								<div class="components-placeholder__label">{ __( 'There is no image generated for the selected image size, so a fallback image size is being used. Learn more.', 'atomic-blocks' ) }</div>
```
Comment (mindctrl) [inline src/blocks/block-post-grid/components/image.js]: This should be `className`.

**Comment by nathanrice** [comment]
Just getting around to testing this. I've cloned the AB repo and switched to the `postGridImages` branch, then ran `npm run build-assets`.

I'm running the Genesis Sample child theme, which registers a new image size of `sidebar-featured` which is 75x75.

Screen Shot 2019-05-09 at 9 35 26 AM

This is what I see in the Genesis Featured Posts widget:

Screen Shot 2019-05-09 at 9 36 16 AM

This is what I see in the Atomic Blocks Page/Post grid block:

Screen Shot 2019-05-09 at 9 35 16 AM

So it looks like something is off. I'm not seeing that image size as an option in the block.

**Comment by mindctrl** [comment]
Thanks for the review @nathanrice! Confirmed what you're seeing. Looks to be a known issue in Gutenberg, but will continue to research and look for alternate methods.

For reference, we currently pull the image sizes from in the editor settings data using the following method: `wp.data.select( 'core/editor' ).getEditorSettings()`.

Will report back with any additional findings.

**Comment by mindctrl** [comment]
Adding the custom image sizes to the `image_size_names_choose` filter makes them available in the dropdown.

**Comment by mindctrl** [comment]
@dreamwhisper If you have time to test this again I'll send treats to Echo! 😆

**Comment by dreamwhisper** [comment]
Haha! Will do @mindctrl.

**Comment by dreamwhisper** [comment]
@mindctrl Nice job with the additional fixes! It is working well for me. I'll tell Echo to expect treats 🐕

**Comment by mindctrl** [comment]
Thanks @dreamwhisper!

**Pull request merged**